core/src/mindustry/game/MapObjectives.java

```diff
@@ -1150,6 +1155,7 @@ public void draw(float scaleFactor){
             if(fetchedRegion == null) setTexture(textureName);
 
             Draw.z(drawLayer);
+            Draw.trans(transform);
```

Comment:
Causing a batch flush every single quad draw call is not acceptable for performance. The vertices need to be manually transformed.